platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/status/SyncPlatformStatusStateMachine.java

```diff
@@ -0,0 +1,203 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.system.status;
+
+import static com.swirlds.logging.LogMarker.EXCEPTION;
+import static com.swirlds.logging.LogMarker.PLATFORM_STATUS;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.notification.NotificationEngine;
+import com.swirlds.common.notification.listeners.PlatformStatusChangeListener;
+import com.swirlds.common.notification.listeners.PlatformStatusChangeNotification;
+import com.swirlds.common.system.status.actions.CatastrophicFailureAction;
+import com.swirlds.common.system.status.actions.DoneReplayingEventsAction;
+import com.swirlds.common.system.status.actions.FallenBehindAction;
+import com.swirlds.common.system.status.actions.FreezePeriodEnteredAction;
+import com.swirlds.common.system.status.actions.PlatformStatusAction;
+import com.swirlds.common.system.status.actions.ReconnectCompleteAction;
+import com.swirlds.common.system.status.actions.SelfEventReachedConsensusAction;
+import com.swirlds.common.system.status.actions.StartedReplayingEventsAction;
+import com.swirlds.common.system.status.actions.StateWrittenToDiskAction;
+import com.swirlds.common.system.status.actions.TimeElapsedAction;
+import com.swirlds.common.system.status.logic.PlatformStatusLogic;
+import com.swirlds.common.system.status.logic.StartingUpStatusLogic;
+import com.swirlds.logging.payloads.PlatformStatusPayload;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A synchronous implementation of the platform status state machine
+ * <p>
+ * NOTE: Processing of {@link PlatformStatusAction}s is not thread-safe. It is assumed that the caller will ensure that
+ * only one thread is calling {@link #processStatusAction(PlatformStatusAction)} at a time.
+ */
+public class SyncPlatformStatusStateMachine implements PlatformStatusStateMachine {
+    private static final Logger logger = LogManager.getLogger(SyncPlatformStatusStateMachine.class);
+
+    /**
+     * A source of time
+     */
+    private final Time time;
+
+    /**
+     * For passing notifications between the platform and the application.
+     */
+    private final NotificationEngine notificationEngine;
+
+    /**
+     * The object containing the state machine logic for the current status
+     */
+    private PlatformStatusLogic currentStatusLogic;
+
+    /**
+     * The current platform status, to be accessed in a thread safe manner
+     */
+    private final AtomicReference<PlatformStatus> currentStatus;
```

Comment:
If we want to follow the component concepts @lpetrovic05 is laying out, this should be in a separate nexus component. We can leave that for later if you prefer - it will be an easy enough change given the current organization. It would replace the `PlatformStatusGetter` functional interface.